src/fiatconnect/index.ts

```diff
@@ -60,6 +65,40 @@ export async function getFiatConnectProviders(
   return providers
 }
 
+/**
+ * Logs in with a FiatConnect provider. Will not attempt to log in if an
+ * unexpired session already exists, unless the `forceLogin` flag is set to `true`.
+ * If the user's wallet is currently locked, will prompt for PIN entry.
+ */
+export async function loginWithFiatConnectProvider(
+  wallet: UnlockableWallet,
+  fiatConnectClient: FiatConnectApiClient,
+  forceLogin: boolean = false
+): Promise<void> {
+  if (fiatConnectClient.isLoggedIn() && !forceLogin) {
+    return
+  }
+
+  const [account] = wallet.getAccounts()
+  if (!wallet.isAccountUnlocked(account)) {
+    await wallet.unlockAccount(account, await getPassword(account), UNLOCK_DURATION)
+  }
+
+  const response = await fiatConnectClient.login()
+  if (!response.isOk) {
+    Logger.error(TAG, `Failure logging in with FiatConnect provider: ${response.error}, throwing`)
```

Comment:
do we want to log the full trace? or is that left to the caller? Also would this affect sentry quota? cc @silasbw